rolling-update-kubernetes/src/main/scala/akka/rollingupdate/kubernetes/AppVersionRevision.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2017-2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.rollingupdate.kubernetes
+
+import akka.util.Version
+import akka.actor.{ ExtendedActorSystem, Extension, ExtensionId, ExtensionIdProvider }
+import akka.cluster.Cluster
+import akka.dispatch.Dispatchers.DefaultBlockingDispatcherId
+import akka.event.Logging
+import akka.rollingupdate.kubernetes.PodDeletionCost.Internal.{ BootstrapStep, Initializing, NotRunning }
+
+import java.util.concurrent.atomic.AtomicReference
+import scala.concurrent.{ ExecutionContext, Future, Promise }
+import scala.util.{ Failure, Success }
+
+final class AppVersionRevision(implicit system: ExtendedActorSystem) extends Extension {
+
+  private val log = Logging(system, classOf[AppVersionRevision])
+  private val configPath = "akka.rollingupdate.kubernetes"
+  private val config = system.settings.config.getConfig(configPath)
+  private val k8sSettings = KubernetesSettings(config)
+
+  private final val startStep = new AtomicReference[BootstrapStep](NotRunning)
+  log.debug("Settings {}", k8sSettings)
+
+  private val versionPromise = Promise[Version]()
+  private final val versionFuture = versionPromise.future
+
+  def getRevision(): Future[Version] = this.versionFuture
+
+  def start(): Unit = {
+    if (startStep.compareAndSet(NotRunning, Initializing)) {
+      Cluster(system).setAppVersionLater(getRevision())
+      implicit val blockingDispatcher: ExecutionContext = system.dispatchers.lookup(DefaultBlockingDispatcherId)
```

Comment:
We only want the `KubernetesApiImpl.apply` to use the blockingDispatcher, and not the Future composition after that. I think it's better to pass in `system` as parameter to `KubernetesApiImpl.apply` and let it lookup the DefaultBlockingDispatcherId